src/main/kotlin/com/sourcegraph/cody/chat/actions/NewChatAction.kt

```diff
@@ -12,8 +12,7 @@ class NewChatAction : DumbAwareEDTAction() {
   }
 
   override fun update(event: AnActionEvent) {
-    val hasActiveAccount = CodyAuthenticationManager.getInstance().hasActiveAccount()
-    event.presentation.isEnabled = hasActiveAccount
+    event.presentation.isEnabled = CodyAccount.hasActiveAccount()
```

Comment:
This `CodyAccount.hasActiveAccount`, we should be using the flag we got from `window/didChangeContext "cody.activated"` instead.

Reply:
I agree in principle, but it is not as simple.
I need to also know account, in particular to be able to check if it's enterprise/pro/free, which in turn I need to properly render LLM selection in edit window.
But now I think maybe we should add additional flag to the model, and avoid this problem in the first place.
It would allow us to remove quite a bit of code on the jetbrains side.

Reply:
OK. Let's work out a practical split to get this landed, because there's a lot of great cleanup here. And we can smooth out the details later.

I agree for `chat/models` returning the models with the flag sounds good.

I tend to disagree in the specific case of NewChatAction, all we need to know is whether we have an active account. But let's land this first and take a fresh look at account type dependencies after this change is done.